README.md

```diff
@@ -1,12 +1,139 @@
+
+## Dependencies (Linux)
+- [OpenModelica](https://openmodelica.org)
+- [Python](https://www.python.org/)
+  - datetime.py
+  - matplotlib.py
+  - monotonic.py
+  - natsort.py
+  - joblib.py
+  - ompython.py
+  - simplejson.py
+  - psutil.py
+- [Optional] Reference Files
+
+
+## Set-Up
+
+- Install or build OpenModelica
+  - [Install instructions](https://openmodelica.org/download/download-linux)
+  - [Build instructions](https://github.com/OpenModelica/OpenModelica#readme)
+  - Make sure `omc` is in your `PATH`
+- Install Python requirements
+  ```bash
+  pip install -r requirements.txt
+  ```
+- Install libraries you want to test
+  - (Optional) Remove already installed libraries
+    ```bash
+    rm -rf ~/.openmodelica/libraries/
+    rm -rf /path/to/OpenModelica/build/lib/omlibraries
+    ```
+  - Install all available libraries.
+    ```bash
+    /path/to/omc .CI/installLibraries.mos
+    ```
+
+    *or*
+  - Install your libraries.
+- Create configs/myConf.json to specify what libraries to test.
+  ```json
+  [
+    {
+      "library":"MyModelicaLibrary",
+      "libraryVersion":"master",
+      "referenceFileExtension":"mat",
+      "referenceFileNameDelimiter":"/",
+      "referenceFileNameExtraName":"$ClassName",
+      "referenceFiles":{
+        "giturl":"https://github.com/myName/MyModelicaLibrary-ref",
+        "destination":"ReferenceFiles/MyModelicaLibrary"
+      },
+      "optlevel":"-Os -march=native"
+    }
+  ]
+  ```
+
+  You can add extra compiler settings
+  ```json
+  "extraCustomCommands":["setCommandLineOptions(\"--std=3.2\");"]
+  ```
+  and more. Check `config/conf.json` for more.
+- If you used `.CI/installLibraries.mos` to test all libraries you'll need to install reference results and set environment variables, see [Reference Results](#reference-results).
+  ```bash
+  export MSLREFERENCE="/path/to/ReferenceFiles/"
+  export REFERENCEFILES="/path/to/OpenModelica/testsuite/ReferenceFiles"
+  export PNLIBREFS="/path/to/ReferenceFiles/PNlib/ReferenceFiles"
+  export THERMOFLUIDSTREAMREFS="/path/to/ReferenceFiles/ThermofluidStream-main-regression/ReferenceData"
+  export THERMOFLUIDSTREAMREFSOM="/path/to/ReferenceFiles/ThermofluidStream-OM-regression/ReferenceData"
+  ```
+
+- Run the library test
+  ```bash
+  ./test.py --noclean configs/myConf.json
+  ```
+  Use `configs/*.json` to specify what to test.
+  The test results are saved in `sqlite3.db`.
+- Generate HTML results
+  ```bash
+  ./report.py configs/myConf.json
+  ```
+- Upload and backup
+  - Upload HTML files somewhere
+  - backup sqlite3.db
+
+### Reference Results
+
+For many available Modelica libraries result files are provide.
+If a library is tested that uses `MSLREFERENCE` or `REFERENCEFILES`
+inside `config/conf.json` and `config/conf-c++.json` one need to
+download the reference files to verify the simulation results.
```

Comment:
But mostly you want to create your own json config and not use those

Reply:
Most likely none besides us will ever use this.

The use case would be to check what the influence of a change in OpenModelica would be for all the libraries without needing to merge it into the master branch. E.g. with the new backend.